1.5/Source/PawnStorages/PawnStorages/Building_PawnStorage.cs

```diff
@@ -61,14 +70,42 @@ public override void Print(SectionLayer layer)
                 pos += StatueOffset.RotatedBy(Rotation);
             }
 
+            s.z *= compAssignable.Props.pawnDrawScaleZ;
+            s.x *= compAssignable.Props.pawnDrawScaleX;
+
             pos += Graphic.DrawOffset(rot);
             pos += StatueOffset;
 
             //Somehow this magically fixes the flipping issue, just keeping it this way.
             mesh.SetUVs(false);
             Printer_Mesh.PrintMesh(layer, Matrix4x4.TRS(pos, Rotation.AsQuat, s), mesh, mat);
+
+            if (ShouldShowOverlay)
+            {
+                pos -= StatueOffset;
+                pos.y = Altitudes.AltitudeFor(AltitudeLayer.MoteOverhead);
+                Printer_Mesh.PrintMesh(layer, Matrix4x4.TRS(pos, Rotation.AsQuat, new Vector3(1,1,1)), OverlayGraphic.MeshAt(Rotation), OverlayGraphic.MatSingle);
+            }
         }
 
         base.Print(layer);
     }
+
+    public ThingOwner GetDirectlyHeldThings()
+    {
+        return storageComp.GetDirectlyHeldThings();
+    }
+
+    public void ReleasePawn(Pawn pawn)
+    {
+        storageComp.ReleaseSingle(this.Map, pawn, true, true);
+
+    }
+
+    public bool NeedsDrop()
+    {
+        CompTickedStorage cmp = GetComp<CompTickedStorage>();
```

Comment:
use the spawn setup pattern like from Building_CommsConsole. Do your get comp at spawn and then use the cached one instead of getting every time.